From 1c986d4842b49016c138ae5ab32f038d0f0c427d Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 29 Sep 2016 15:40:43 -0700
Subject: [PATCH] Check lmc_mode32 for returning the correct alternative
 package

Some 80xx parts are not fused out properly so check mio_fus_dat2.lmc_mode32
to determine the alternate package.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>

---
 board/cavium/thunderx/thunderx.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/board/cavium/thunderx/thunderx.c b/board/cavium/thunderx/thunderx.c
index 54177b8afd..0b626d5142 100644
--- a/board/cavium/thunderx/thunderx.c
+++ b/board/cavium/thunderx/thunderx.c
@@ -113,8 +113,16 @@ void reset_cpu(ulong addr)
 bool alternate_pkg(void)
 {
 	u64 val = readq(CAVM_MIO_FUS_DAT2);
+	int altpkg;
 
-	return (val >> 22) & 0x3;
+	altpkg = (val >> 22) & 0x3;
+
+	/* Figure out alternative pkg by reading chip_id
+	   or lmc_mode32 on 81xx */
+	if (CAVIUM_IS_MODEL(CAVIUM_CN81XX)
+	    && (altpkg || ((val >> 30) & 0x1)))
+		return 2;
+	return altpkg;
 }
 
 /**
